src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -9,14 +9,187 @@ import is.hail.utils._
+    var i = 0
+    while (i < ref.length) {
+      val c = ref(i)
+      (c: @switch) match {
+        case 'A' | 'C' | 'G' | 'T' | 'N' |
+             'a' | 'c' | 'g' | 't' | 'n' =>
+        case _ =>
+          return true
+      }
+      i += 1
+    }
+    false
+  }
+
+  def symbolicAlt(alt: String): Boolean = {
+    assert(alt.length > 0)
+    val f = alt(0)
+    val l = alt(alt.length - 1)
+    if (f == '.' || l == '.' || (f == '<' && l == '>'))
+      return true
+
+    var i = 0
+    while (i < alt.length) {
+      val c = alt(i)
+      if (c == '[' || c == ']')
+        return true
+      i += 1
+    }
+    false
+  }
+}
+
+final class VCFLine(line: String) {
+  var pos: Int = 0
+
+  val abs = new ArrayBuilder[String]
+
+  def parseString(): String = {
+    val start = pos
+    while (pos < line.length && line(pos) != '\t')
+      pos += 1
+    val end = pos
+    line.substring(start, end)
+  }
+
+  def parseInt(): Int = {
+    var c: Char = 0
+
+    if (!(pos < line.length && line(pos) != '\t'))
+      fatal("empty integer field")
+    var v = VCFLine.numericValue(line(pos))
+    pos += 1
+    while (pos < line.length && line(pos) != '\t') {
+      v = v * 10 + VCFLine.numericValue(line(pos))
+      pos += 1
+    }
+    v
+  }
+
+  def skipField(): Unit = {
+    while (pos < line.length && line(pos) != '\t')
+      pos += 1
+  }
+
+  def parseStringInArray(): String = {
+    val start = pos
+    while (pos < line.length && line(pos) != '\t' && line(pos) != ',')
+      pos += 1
+    val end = pos
+    line.substring(start, end)
+  }
+
+  // leaves result in abs
+  def parseAltAlleles(): Unit = {
+    assert(abs.size == 0)
+
+    // . means no alternate alleles
+    if (pos < line.length
+      && line(pos) == '.'
+      && (pos + 1 == line.length || line(pos + 1) == '\t'))
+      return
+
+    abs += parseStringInArray()
+    while (pos < line.length && line(pos) != '\t') {
+      pos += 1 // comma
+      abs += parseStringInArray()
+    }
+  }
+
+  def nextField(): Unit = {
+    if (pos == line.length)
+      fatal("unexpected end of line")
+    pos += 1 // tab
+  }
+
+  // return false if it should be filtered
+  def parseAddVariant(rvb: RegionValueBuilder): Boolean = {
+    assert(pos == 0)
+
+    if (line.isEmpty || line(0) == '#')
+      return false
+
+    // CHROM (contig)
+    val contig = parseString()
+    nextField()
+
+    // POS (start)
+    val start = parseInt()
+    nextField()
+
+    skipField() // ID
+    nextField()
+
+    // REF
+    val ref = parseString()
+    if (VCFLine.invalidRef(ref)) {
+      warn(s"skipping variant with invalid REF field: $ref")
+      return false
+    }
+    nextField()
+
+    // ALT
+    parseAltAlleles()
+
+    var i = 0
+    while (i < abs.length) {
+      val alt = abs(i)
+      if (alt.isEmpty)
+        fatal("empty alternate allele")
+      if (VCFLine.symbolicAlt(alt)) {
+        warn(s"skipping variant with symbolic alternate allele in ALT field: $alt")
```

Comment:
same as above.

Reply:
We could actually use accumulators here, since we need to consume the RDD to compute the partition bounds.

Reply:
Hmm.  Seems better than nothing tho, right?

I always say, we shouldn't even have to do the initial scan!  I also want a `sorted=True` fast path which verifies sortedness, which would break this.  That should always work if the contigs are sorted and there are no duplicate loci.

Reply:
Happy to see this in another PR as long as we don't forget

Reply:
Happy to see this in another PR as long as we don't forget